File: truffle/SimpleStorage/SimpleStorageService.vb
Imports System
Imports System.Threading.Tasks
Imports System.Collections.Generic
Imports System.Numerics
Imports Nethereum.Hex.HexTypes
Imports Nethereum.ABI.FunctionEncoding.Attributes
Imports Nethereum.Web3
Imports Nethereum.RPC.Eth.DTOs
Imports Nethereum.Contracts.CQS
Imports Nethereum.Contracts.ContractHandlers
Imports Nethereum.Contracts
Imports System.Threading
Imports Truffle.Contracts.SimpleStorage.ContractDefinition
Namespace Truffle.Contracts.SimpleStorage


    Public Partial Class SimpleStorageService
    
    
        Public Shared Function DeployContractAndWaitForReceiptAsync(ByVal web3 As Nethereum.Web3.Web3, ByVal simpleStorageDeployment As SimpleStorageDeployment, ByVal Optional cancellationTokenSource As CancellationTokenSource = Nothing) As Task(Of TransactionReceipt)
        
            Return web3.Eth.GetContractDeploymentHandler(Of SimpleStorageDeployment)().SendRequestAndWaitForReceiptAsync(simpleStorageDeployment, cancellationTokenSource)
        
        End Function
         Public Shared Function DeployContractAsync(ByVal web3 As Nethereum.Web3.Web3, ByVal simpleStorageDeployment As SimpleStorageDeployment) As Task(Of String)
        
            Return web3.Eth.GetContractDeploymentHandler(Of SimpleStorageDeployment)().SendRequestAsync(simpleStorageDeployment)
        
        End Function
        Public Shared Async Function DeployContractAndGetServiceAsync(ByVal web3 As Nethereum.Web3.Web3, ByVal simpleStorageDeployment As SimpleStorageDeployment, ByVal Optional cancellationTokenSource As CancellationTokenSource = Nothing) As Task(Of SimpleStorageService)
        
            Dim receipt = Await DeployContractAndWaitForReceiptAsync(web3, simpleStorageDeployment, cancellationTokenSource)
            Return New SimpleStorageService(web3, receipt.ContractAddress)
        
        End Function
    
        Protected Property Web3 As Nethereum.Web3.IWeb3
        
        Public Property ContractHandler As ContractHandler
        
        Public Sub New(ByVal web3 As Nethereum.Web3.Web3, ByVal contractAddress As String)
            Web3 = web3
            ContractHandler = web3.Eth.GetContractHandler(contractAddress)
        End Sub
    
        Public Sub New(ByVal web3 As Nethereum.Web3.IWeb3, ByVal contractAddress As String)
            Web3 = web3
            ContractHandler = web3.Eth.GetContractHandler(contractAddress)
        End Sub
    
        Public Function ReadQueryAsync(ByVal readFunction As ReadFunction, ByVal Optional blockParameter As BlockParameter = Nothing) As Task(Of BigInteger)
        
            Return ContractHandler.QueryAsync(Of ReadFunction, BigInteger)(readFunction, blockParameter)
        
        End Function

        
        Public Function ReadQueryAsync(ByVal Optional blockParameter As BlockParameter = Nothing) As Task(Of BigInteger)
        
            return ContractHandler.QueryAsync(Of ReadFunction, BigInteger)(Nothing, blockParameter)
        
        End Function



        Public Function WriteRequestAsync(ByVal writeFunction As WriteFunction) As Task(Of String)
                    
            Return ContractHandler.SendRequestAsync(Of WriteFunction)(writeFunction)
        
        End Function

        Public Function WriteRequestAndWaitForReceiptAsync(ByVal writeFunction As WriteFunction, ByVal Optional cancellationToken As CancellationTokenSource = Nothing) As Task(Of TransactionReceipt)
        
            Return ContractHandler.SendRequestAndWaitForReceiptAsync(Of WriteFunction)(writeFunction, cancellationToken)
        
        End Function

        
        Public Function WriteRequestAsync(ByVal [newValue] As BigInteger) As Task(Of String)
        
            Dim writeFunction = New WriteFunction()
                writeFunction.NewValue = [newValue]
            
            Return ContractHandler.SendRequestAsync(Of WriteFunction)(writeFunction)
        
        End Function

        
        Public Function WriteRequestAndWaitForReceiptAsync(ByVal [newValue] As BigInteger, ByVal Optional cancellationToken As CancellationTokenSource = Nothing) As Task(Of TransactionReceipt)
        
            Dim writeFunction = New WriteFunction()
                writeFunction.NewValue = [newValue]
            
            Return ContractHandler.SendRequestAndWaitForReceiptAsync(Of WriteFunction)(writeFunction, cancellationToken)
        
        End Function
    
    End Class

End Namespace

File: truffle/SimpleStorage/SimpleStorageService.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Numerics;
using Nethereum.Hex.HexTypes;
using Nethereum.ABI.FunctionEncoding.Attributes;
using Nethereum.Web3;
using Nethereum.RPC.Eth.DTOs;
using Nethereum.Contracts.CQS;
using Nethereum.Contracts.ContractHandlers;
using Nethereum.Contracts;
using System.Threading;
using Truffle.Contracts.SimpleStorage.ContractDefinition;

namespace Truffle.Contracts.SimpleStorage
{
    public partial class SimpleStorageService
    {
        public static Task<TransactionReceipt> DeployContractAndWaitForReceiptAsync(Nethereum.Web3.Web3 web3, SimpleStorageDeployment simpleStorageDeployment, CancellationTokenSource cancellationTokenSource = null)
        {
            return web3.Eth.GetContractDeploymentHandler<SimpleStorageDeployment>().SendRequestAndWaitForReceiptAsync(simpleStorageDeployment, cancellationTokenSource);
        }

        public static Task<string> DeployContractAsync(Nethereum.Web3.Web3 web3, SimpleStorageDeployment simpleStorageDeployment)
        {
            return web3.Eth.GetContractDeploymentHandler<SimpleStorageDeployment>().SendRequestAsync(simpleStorageDeployment);
        }

        public static async Task<SimpleStorageService> DeployContractAndGetServiceAsync(Nethereum.Web3.Web3 web3, SimpleStorageDeployment simpleStorageDeployment, CancellationTokenSource cancellationTokenSource = null)
        {
            var receipt = await DeployContractAndWaitForReceiptAsync(web3, simpleStorageDeployment, cancellationTokenSource);
            return new SimpleStorageService(web3, receipt.ContractAddress);
        }

        protected Nethereum.Web3.IWeb3 Web3{ get; }

        public ContractHandler ContractHandler { get; }

        public SimpleStorageService(Nethereum.Web3.Web3 web3, string contractAddress)
        {
            Web3 = web3;
            ContractHandler = web3.Eth.GetContractHandler(contractAddress);
        }

        public SimpleStorageService(Nethereum.Web3.IWeb3 web3, string contractAddress)
        {
            Web3 = web3;
            ContractHandler = web3.Eth.GetContractHandler(contractAddress);
        }

        public Task<BigInteger> ReadQueryAsync(ReadFunction readFunction, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<ReadFunction, BigInteger>(readFunction, blockParameter);
        }

        
        public Task<BigInteger> ReadQueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<ReadFunction, BigInteger>(null, blockParameter);
        }

        public Task<string> WriteRequestAsync(WriteFunction writeFunction)
        {
             return ContractHandler.SendRequestAsync(writeFunction);
        }

        public Task<TransactionReceipt> WriteRequestAndWaitForReceiptAsync(WriteFunction writeFunction, CancellationTokenSource cancellationToken = null)
        {
             return ContractHandler.SendRequestAndWaitForReceiptAsync(writeFunction, cancellationToken);
        }

        public Task<string> WriteRequestAsync(BigInteger newValue)
        {
            var writeFunction = new WriteFunction();
                writeFunction.NewValue = newValue;
            
             return ContractHandler.SendRequestAsync(writeFunction);
        }

        public Task<TransactionReceipt> WriteRequestAndWaitForReceiptAsync(BigInteger newValue, CancellationTokenSource cancellationToken = null)
        {
            var writeFunction = new WriteFunction();
                writeFunction.NewValue = newValue;
            
             return ContractHandler.SendRequestAndWaitForReceiptAsync(writeFunction, cancellationToken);
        }
    }
}
